base/dcp_client_test.go

```diff
@@ -333,8 +327,10 @@ func TestResumeStoppedFeed(t *testing.T) {
 		if bytes.HasPrefix(event.Key, []byte(t.Name())) {
 			count := atomic.AddUint64(&mutationCount, 1)
 			if count > 5000 {
-				err := dcpClient.Close()
-				assert.NoError(t, err)
+				// use go routine to not deadlock, since this callback will prevent a close from happening
```

Comment:
If DCP clients can't invoke Close() within the callback without spawning a separate goroutine, I'm not sure that a blocking external Close() is the pattern we want?  Seems like an easy way to introduce deadlocks.

Reply:
See the above comment, but in all cases here we are doing a synchronous close so I wonder if a public helper function is worthwhile.